router/dataplane.go

```diff
@@ -3773,11 +3761,16 @@ func addEndhostPort(dst *net.IPAddr) (res *net.UDPAddr) {
 // the scion.Raw path.
 // @ requires  acc(s.Mem(rawPkt), R00)
 // @ requires  s.HasOneHopPath(rawPkt)
+// @ requires  sl.Bytes(rawPkt, 0, len(rawPkt))
 // @ preserves buffer != nil && buffer.Mem()
-// @ preserves sl.Bytes(rawPkt, 0, len(rawPkt))
+// @ ensures   sl.Bytes(rawPkt, 0, len(rawPkt))
 // @ ensures   acc(s.Mem(rawPkt), R00)
 // @ ensures   res != nil ==> res.ErrorMem()
 // @ decreases
+// Contracts for IO-sepc
+// @ requires s.EqPathType(rawPkt)
+// @ requires !slayers.IsSupportedPkt(rawPkt)
```

Comment:
please put this together with the other preconditions (you may add a comment saying that these preconditions have to do with IO spec)

Reply:
Can you double-check to see if it is better now?